src/libsystemd-network/sd-dhcp6-client.c

```diff
@@ -188,17 +188,18 @@ int sd_dhcp6_client_set_mac(
 
         assert_return(client, -EINVAL);
         assert_return(addr, -EINVAL);
-        assert_return(addr_len > 0 && addr_len <= MAX_MAC_ADDR_LEN, -EINVAL);
```

Comment:
Please keep the condition `addr_len <= MAX_MAX_ADDR_LEN`.

Reply:
I restored it.